### DISCUSSION TITLE
Info Voltronic King UNO Parallel

### DISCUSSION THREAD

**Discussion by saviothecnic** [syssi/esphome-pipsolar#93] [General]
I'm trying your code
I also made a video where I explain how everything connects
So that even less experienced people know how to use your program.

The code is fine and is talking to my Voltronic KING UNO
I believe it is a PI30 MK protocol

But I'm not sure if your code reads my two inverters
in Parallel I see little data
Question do you have a list of sensors that I can activate
and what commands can I send to the inverter?
So I'll be happy to try your code for you
on my King inverters in parallel and possibly
understand if there is anything missing to add

![image](https://github.com/syssi/esphome-pipsolar/assets/6697807/a0c5a4ab-8486-44bd-9c54-03cf11cdb9ad)

![image](https://github.com/syssi/esphome-pipsolar/assets/6697807/1a4c95a4-f495-4653-8877-f16dcd1be35b)

https://youtu.be/C9UTF_FwQco?si=0R-50vS0L3Lzcr1M

**Comment by saviothecnic** [comment]
I attach the yaml that I am using but I think it is for a Voltronic Max inverter, can anyone help me add the sensors
for the voltronic King and eliminate the sensors for MPPT here I have two the King only has One

[Voltronic-Rs232.yaml.txt](https://github.com/syssi/esphome-pipsolar/files/13523841/Voltronic-Rs232.yaml.txt)

For example, I would need the sensor to know the operating mode as per page 4 of the Z option manual
https://github.com/jblance/mpp-solar/blob/master/docs/protocols/PI30_PIP-GK_MK-Protocol.pdf

Another question should I do an automation with node red
but I didn't understand which topic I should send to mqtt
Can you help me to change these three options?

![image](https://github.com/syssi/esphome-pipsolar/assets/6697807/afa74c43-55ac-4105-aab9-509551bd193f)

**Reply by saviothecnic** [reply]
So no one can help me?

I also migrated to Esp32 Athom which as HW
It's easier as you can see from the video

https://youtu.be/3eekwPefVII?si=TYHyqv5P1vMwPKst
[https://youtu.be/3eekwPefVII?si=TYHyqv5P1vMwPKst](https://github.com/syssi/esphome-pipsolar/discussions/url) 

Codice Yaml EspHome By Savio 

`substitutions:
  name: voltronic
  friendly_name: Voltronic Inverter King UNO RS-232

esphome:
  name: ${name}
  comment: "Voltronic King UNO RS485 PIP30 protocol"
  platformio_options:
    upload_speed: 115200
    build_flags: 
      - -DCONFIG_ARDUINO_LOOP_STACK_SIZE=32768

esp32:
  board: m5stack-atom

external_components:
  - source: github://syssi/esphome-pipsolar@pip8048
    refresh: 0s

wifi:
  networks:
    - ssid: !secret wifi_ssid
      password: !secret wifi_password
    - ssid: !secret wifi_ssid_2
      password: !secret wifi_password_2

  manual_ip:
    static_ip: 192.168.1.44
    gateway: 192.168.1.254
    subnet: 255.255.255.0
    dns1: 8.8.8.8
    dns2: 192.168.1.254
 
web_server:
  port: 80
  
ota:

logger:
  baud_rate: 0

mqtt:
   broker: !secret mqtt_host
   username: !secret mqtt_username
   password: !secret mqtt_password
   id: mqtt_client

uart:
  id: uart_bus
  baud_rate: 2400
  rx_pin: GPIO22
  tx_pin: GPIO19

pipsolar:
  uart_id: uart_bus
  id: inverter0

sensor:
  # QPIRI
  - platform: pipsolar
    pipsolar_id: inverter0
    grid_rating_voltage:
      name: "${name} grid_rating_voltage"
    grid_rating_current:
      name: "${name} grid_rating_current"
    ac_output_rating_voltage:
      name: "${name} ac_output_rating_voltage"
    ac_output_rating_frequency:
      name: "${name} ac_output_rating_frequency"
    ac_output_rating_current:
      name: "${name} ac_output_rating_current"
    ac_output_rating_apparent_power:
      name: "${name} ac_output_rating_apparent_power"
    ac_output_rating_active_power:
      name: "${name} ac_output_rating_active_power"
    battery_rating_voltage:
      name: "${name} battery_rating_voltage"
    battery_recharge_voltage:
      name: "${name} battery_recharge_voltage"
    battery_under_voltage:
      name: "${name} battery_under_voltage"
    battery_bulk_voltage:
      name: "${name} battery_bulk_voltage"
    battery_float_voltage:
      name: "${name} battery_float_voltage"
    battery_type:
      name: "${name} battery_type"
    current_max_ac_charging_current:
      name: "${name} current_max_ac_charging_current"
    current_max_charging_current:
      name: "${name} current_max_charging_current"
    input_voltage_range:
      name: "${name} input_voltage_range"
    output_source_priority:
      name: "${name} output_source_priority"
    charger_source_priority:
      name: "${name} charger_source_priority"
    parallel_max_num:
      name: "${name} parallel_max_num"
    machine_type:
      name: "${name} machine_type"
    topology:
      name: "${name} topology"
    output_mode:
      name: "${name} output_mode"
    battery_redischarge_voltage:
      name: "${name} battery_redischarge_voltage"
    pv_ok_condition_for_parallel:
      name: "${name} pv_ok_condition_for_parallel"
    pv_power_balance:
      name: "${name} pv_power_balance"

  # QPIGS
  - platform: pipsolar
    pipsolar_id: inverter0
    grid_voltage:
      name: "${name} grid_voltage"
    grid_frequency:
      name: "${name} grid_frequency"
    ac_output_voltage:
      name: "${name} ac_output_voltage"
    ac_output_frequency:
      name: "${name} ac_output_frequency"
    ac_output_apparent_power:
      name: "${name} ac_output_apparent_power"
    ac_output_active_power:
      name: "${name} ac_output_active_power"
    output_load_percent:
      name: "${name} output_load_percent"
    bus_voltage:
      name: "${name} bus_voltage"
    battery_voltage:
      name: "${name} battery_voltage"
    battery_charging_current:
      name: "${name} battery_charging_current"
    battery_capacity_percent:
      name: "${name} battery_capacity_percent"
    inverter_heat_sink_temperature:
      name: "${name} inverter_heat_sink_temperature"
    pv1_input_current:
      name: "${name} pv1_input_current"
    pv1_input_voltage:
      name: "${name} pv1_input_voltage"
    battery_voltage_scc:
      name: "${name} battery_voltage_scc"
    battery_discharge_current:
      name: "${name} battery_discharge_current"
    battery_voltage_offset_for_fans_on:
      name: "${name} battery_voltage_offset_for_fans_on"
    eeprom_version:
      name: "${name} eeprom_version"
    pv1_charging_power:
      name: "${name} pv1_charging_power"

text_sensor:
  - platform: pipsolar
    pipsolar_id: inverter0
    device_mode:
      name: "${name} device_mode"
#    last_qpigs:
#      name: "${name} last_qpigs"
#    last_qpiri:
#      name: "${name} last_qpiri"
#    last_qmod:
#      name: "${name} last_qmod"
#    last_qflag:
#      name: "${name} last_qflag"

binary_sensor:
  - platform: pipsolar
    pipsolar_id: inverter0
    add_sbu_priority_version:
      name: "${name} add_sbu_priority_version"
    configuration_status:
      name: "${name} configuration_status"
    scc_firmware_version:
      name: "${name} scc_firmware_version"
    load_status:
      name: "${name} load_status"
    battery_voltage_to_steady_while_charging:
      name: "${name} battery_voltage_to_steady_while_charging"
    charging_status:
      name: "${name} charging_status"
    scc_charging_status:
      name: "${name} scc_charging_status"
    ac_charging_status:
      name: "${name} ac_charging_status"
    charging_to_floating_mode:
      name: "${name} charging_to_floating_mode"
    switch_on:
      name: "${name} switch_on"
#    dustproof_installed:
#      name: "${name} dustproof_installed"
    silence_buzzer_open_buzzer:
      name: "${name} silence_buzzer_open_buzzer"
    overload_bypass_function:
      name: "${name} overload_bypass_function"
    lcd_escape_to_default:
      name: "${name} lcd_escape_to_default"
    overload_restart_function:
      name: "${name} overload_restart_function"
    over_temperature_restart_function:
      name: "${name} over_temperature_restart_function"
#    backlight_on:
#      name: "${name} backlight_on"

switch:
  - platform: pipsolar
    pipsolar_id: inverter0
    output_source_priority_utility:
      name: "${name} output_source_priority_utility"
    output_source_priority_solar:
      name: "${name} output_source_priority_solar"
    output_source_priority_battery:
      name: "${name} output_source_priority_battery"
    input_voltage_range:
      name: "${name} input_voltage_range"
    pv_ok_condition_for_parallel:
      name: "${name} pv_ok_condition_for_parallel"
    pv_power_balance:
      name: "${name} pv_power_balance"

output:
  - platform: pipsolar
    pipsolar_id: inverter0
    battery_recharge_voltage:
      id: inverter0_battery_recharge_voltage_out

  - platform: pipsolar
    pipsolar_id: inverter0
    battery_bulk_voltage:
      id: inverter0_battery_bulk_voltage_out

select:
  - platform: pipsolar
    pipsolar_id: inverter0
    output_source_priority:
      id: inverter0_output_source_priority_select
      name: inverter0_output_source_priority_select
      optionsmap:
        "Utility first": "POP00"
        "Solar only": "POP01"
        "Solar Battery Utility": "POP02"
      statusmap:
        "0": "Utility first"
        "1": "Solar only"
        "2": "Solar Battery Utility"

  - platform: pipsolar
    pipsolar_id: inverter0
    charging_discharging_control:
      name: "${name} charging discharging control"
      optionsmap:
        "111": "PBATCD111"
        "011": "PBATCD011"
        "101": "PBATCD101"
        "110": "PBATCD110"
        "010": "PBATCD010"
        "100": "PBATCD100"
        "001": "PBATCD001"
        "000": "PBATCD000"
      statusmap:
        "111": "111"
        "011": "011"
        "101": "101"
        "110": "110"
        "010": "010"
        "100": "100"
        "001": "001"
        "000": "000"

  - platform: pipsolar
    pipsolar_id: inverter0
    charger_source_priority:
      name: "${name} charger_source_priority_select"
      optionsmap:
        "Utility first": "PCP00"
        "Solar first": "PCP01"
        "Solar and utility": "PCP02"
        "Solar charging only": "PCP03"
      statusmap:
        "0": "Utility first"
        "1": "Solar first"
        "2": "Solar and utility"
        "3": "Solar charging only"

  - platform: pipsolar
    pipsolar_id: inverter0
    current_max_ac_charging_current:
      name: "${name} current_max_ac_charging_current_select"
      optionsmap:
        "2A": "MUCHGC0002"
        "10A": "MUCHGC0010"
        "20A": "MUCHGC0020"
        "30A": "MUCHGC0030"
        "40A": "MUCHGC0040"
        "50A": "MUCHGC0050"
        "60A": "MUCHGC0060"
        "70A": "MUCHGC0070"
        "80A": "MUCHGC0080"
        "90A": "MUCHGC0090"
        "100A": "MUCHGC0100"
        "110A": "MUCHGC0110"
        "120A": "MUCHGC0120"
      statusmap:
        "2": "2A"
        "10": "10A"
        "20": "20A"
        "30": "30A"
        "40": "40A"
        "50": "50A"
        "60": "60A"
        "70": "70A"
        "80": "80A"
        "90": "90A"
        "100": "100A"
        "110": "110A"
        "120": "120A"

  - platform: pipsolar
    pipsolar_id: inverter0
    current_max_charging_current:
      name: "${name} current_max_charging_current_select"
      optionsmap:
        "10A": "MCHGC010"
        "20A": "MCHGC020"
        "30A": "MCHGC030"
        "40A": "MCHGC040"
        "50A": "MCHGC050"
        "60A": "MCHGC060"
        "70A": "MCHGC070"
        "80A": "MCHGC080"
        "90A": "MCHGC090"
        "100A": "MCHGC100"
        "110A": "MCHGC110"
        "120A": "MCHGC120"
      statusmap:
        "10": "10A"
        "20": "20A"
        "30": "30A"
        "40": "40A"
        "50": "50A"
        "60": "60A"
        "70": "70A"
        "80": "80A"
        "90": "90A"
        "100": "100A"
        "110": "110A"
        "120": "120A"`
https://youtu.be/3eekwPefVII?si=TYHyqv5P1vMwPKst
[https://youtu.be/3eekwPefVII?si=TYHyqv5P1vMwPKst](url)

**Reply by Xim1994** [reply]
Hello @saviothecnic if I understood correct you could do all the setup with just one atom device? It worked for you? I'm trying to do the same, integrate my Voltronic King inverter to my Home Assistant.
